File: frontend/lib/features/leave_management/domain/entities/leave_type.dart
import 'package:equatable/equatable.dart';

class LeaveType extends Equatable {
  final int id;
  final String name;
  final String code;
  final bool isPaid;

  const LeaveType({
    required this.id,
    required this.name,
    required this.code,
    required this.isPaid,
  });

  @override
  List<Object?> get props => [id, name, code, isPaid];
}

File: backends/business/app/Models/LeaveType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class LeaveType extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'code',
        'default_days',
        'is_paid',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'is_paid' => 'boolean',
            'is_active' => 'boolean',
        ];
    }

    /**
     * Get leave requests for this type.
     */
    public function leaveRequests()
    {
        return $this->hasMany(LeaveRequest::class);
    }
}
